### PR TITLE
controller crashes if AS3 schema cannot be fetched #1040

### PR DESCRIPTION
**Pull request by srivastavnitin24**
Problem: Controller crashes if AS3 schema cannot be fetched.

Solution: Handling responses to fetch the latest AS3 Schema and implement the fallback mechanism.

Build : nitin24/as3schemavalidation:1.0

Affected-branches: Master, 1.11-stable

### CODE REVIEW

**Comment by ssurenr** [comment]
Build : nitin24/as3schemavalidation:1.0

File: cmd/k8s-bigip-ctlr/main.go [hunk 1]
```diff
@@ -780,12 +781,15 @@ func main() {
...
-	res, resErr := http.Get(as3SchemaLatestUrl)
+	res, resErr := http.Get(as3SchemaLatestURL)
 	if resErr != nil {
 		log.Debugf("error while fetching latest as3 schema : %v", resErr)
 	}
-
-	if res.StatusCode == http.StatusOK {
+	if res == nil {
+		appMgr.As3SchemaFlag = true
+		log.Debugf("unable to fetch the latest AS3 schema : validating AS3 schema with %v", as3SchemaFileName)
```
Comment (ssurenr) [inline cmd/k8s-bigip-ctlr/main.go]:
```suggestion
		log.Errorf("unable to fetch the latest AS3 schema : validating AS3 schema with %v", as3SchemaFileName)
```

File: cmd/k8s-bigip-ctlr/main.go [hunk 2]
```diff
@@ -780,12 +781,15 @@ func main() {
...
+		appMgr.As3SchemaLatest = appMgr.SchemaLocalPath + as3SchemaFileName
+	} else if res.StatusCode == http.StatusOK && res != nil {
```
Comment (ssurenr) [inline cmd/k8s-bigip-ctlr/main.go]: This else is required? You can keep the if condition outside the else block.

File: cmd/k8s-bigip-ctlr/main.go (hunk 1 above)
Comment (srivastavnitin24) [reply cmd/k8s-bigip-ctlr/main.go]: done

**Pull request closed without merging**